tests/optimisation_test_workflow/Loop_test_script_Optimization.py

```diff
@@ -0,0 +1,83 @@
+from typing import Dict
+import yaml
+import math as m
+from archemist.core.persistence.object_factory import OptimizationFactory
+from archemist.core.optimisation.bayesopt_optimiser import BayesOptOptimizer
+from archemist.core.persistence.yaml_handler import YamlHandler
+from mongoengine import connect
+from pathlib import Path
+import random
+import pandas as pd
+
+class Optimization_test:
+    def __init__(self) -> None:
+        cwd_path = Path.cwd()
+        workflow_dir = Path.joinpath(cwd_path, "tests/optimisation_test_workflow")
+        self._config_file = Path.joinpath(workflow_dir, "config_files/optimization_config.yaml")
+        self._config_dict = YamlHandler.loadYamlFile(self._config_file)
+        self._batch_size = self._config_dict['optimizer']['batch_size']
+        self._optimization_state = OptimizationFactory.create_from_dict(self._config_dict['optimizer'])
+        self._optimizer = BayesOptOptimizer(self._optimization_state,  self._config_dict)
+        values_from_optimizer = self._optimizer.generate_batch()
+        print(values_from_optimizer)
+
+        for iter in range(100):
+            self._random_values_dict = {}
+            self._random_values_dict = self.generate_random_values()
+
+            self.params = {}
+            keys = ['dye_A', 'dye_B', 'water']
+            for key in keys:
+                self.params[key] = None
+            self._find_parameters(self._random_values_dict)
+            self.update_function()
+            print(f'iteration_{iter+1}: \n',self._random_values_dict)
+            result_data_pd = pd.DataFrame(self._random_values_dict)
+            self._optimizer.update_model(result_data_pd)
+        
+        for i in range(10):
+            values_from_optimizer = self._optimizer.generate_batch()
+            print(f'batch_{i+1}: \n', values_from_optimizer)
```

Comment:
how the optimiser is being tested as I don't see anything being optimiser. The first for loop generate random values and update the model, while the second loop generate batches. The correct way should be something like:

for i in range(100):
  if i == 1:
    batch_values = generate_random_values()
  fitness = calculate_fitness_function(batch_values)
  batch_values = generate_batch()

in the end of the for loop, your fitness value should be the min of your fitness function